molgenis/molgenis-emx2
# feat(directory): molgenis-emx2-directory-client

What are the main changes you did:
Created a new tool: molgenis-emx2-directory-client, which is also known as the EMX2 Publish script for BBMRI-ERIC, which combines data from the staging areas, add some extra fields and adds the data to the ERIC tables which are used in the Directory App. 
This is initial a copy of molgenis-py-bbmri-eric 1.18.1, which is the EMX1 version of the Publish script.

how to test:
- On a test-server, create a database based on the biobank_directory template (no demo data)
- Create one or more databases based on the biobank_directory_staging template (with demo data)
- Create a script by copy paste the "dev.py"
- Add the latest version on testPyPI of molgenis-emx2-directory-client to the requirements
- Run the script
- Check that the data from the staging area(s) is in the directory database 

todo:
- [x] Adjust the project setup files to get it working in EMX2
~~- [ ] Adjust README~~ => separate [ticket](https://github.com/molgenis/GCC/issues/820)
- [x] Replace molgenis-py-client with molgenis-emx2-pyclient
~~- [ ] updated docs in case of new feature~~ => separate [ticket](https://github.com/molgenis/GCC/issues/820) 

~~Optional todo:~~ separate [ticket](https://github.com/molgenis/GCC/issues/820)
- [x] test coverage is not 76%, but is not checked before final merge, anyway perhaps some extra tests could be added
- [x] Refactoring: Some code has become redundant (f.e. convert_to_staging as tables names are now equal between ERIC and staging areas), some methods can be rewritten to make them more efficient/logical
